src/modules/Lessons/9_LessonAPIIntegration/assignment/Dashboard.js

```diff
@@ -0,0 +1,135 @@
+import React from "react";
+import axios from "axios";
+import ErrorMsg from "./ErrorMsg";
+import Results from "./Results";
+import LoaderImage from "./LoaderImage";
+import SearchBox from "./SearchBox";
+
+import './Dashboard.css'
+import ResultNotification from "./ResultNotification";
+
+const APP_STATE = {
+  IS_LOADING: 'IS_LOADING',
+  IS_ERRORED: 'IS_ERRORED',
+  IS_READY: 'IS_READY'
+};
+
+const DASHBOARD_TYPE = {
+  USERS: 'USERS',
+  REPOS: 'REPOS',
+};
+
+const API_URL = `https://api.github.com/users`;
+
+class Dashboard extends React.Component {
+  constructor(props) {
+    super(props);
+    let dashboardType = DASHBOARD_TYPE.USERS;
+    let apiURL = API_URL;
+    if (props.match && props.match.params.userId) {
+      dashboardType = DASHBOARD_TYPE.REPOS;
+      apiURL = `${API_URL}/${props.match.params.userId}/repos`;
+
+    }
+
+
+    this.state = {
+      appState: APP_STATE.IS_LOADING,
+      results: [],
+      dashboard: dashboardType,
+      query: '',
+      error: '',
+      apiUrl: apiURL,
+    }
+  }
+
+  componentWillMount() {
+    this.fetchData(this.onSuccess, this.onError);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.updateDashBoardState(nextProps.match);
+    setTimeout(()=>{
```

Comment:
@InamTaj I don't know what other workaround I should have used instead of this shitty approach. Maybe change the architecture a bit for this assignment implementation. WDYT?